source/renderer/app/components/staking/widgets/ThumbPool.js

```diff
@@ -1,162 +1,85 @@
-  onSelect: Function,
+  onSelect?: (poolId: string) => void,
+  selectOnClick?: boolean,
   showWithSelectButton?: boolean,
-  showSelected?: boolean,
   stakePool: StakePool,
-  isSelected?: ?Function,
   containerClassName: string,
   numberOfRankedStakePools: number,
   disabledStakePoolId: ?string,
-};
-
-type State = {
-  top: number,
-  left: number,
-};
-
-@observer
-export class ThumbPool extends Component<Props, State> {
-  state = {
-    top: 0,
-    left: 0,
-  };
-
-  hoverWait: TimeoutID;
-
-  handleHover = (stakePoolId: string) => {
-    clearTimeout(this.hoverWait);
-    this.hoverWait = setTimeout(() => {
-      if (this.props.onHover) this.props.onHover(stakePoolId);
-    }, STAKE_POOL_TOOLTIP_HOVER_WAIT);
-  };
-
-  handleClose = (stakePoolId: string) => {
-    clearTimeout(this.hoverWait);
-    this.props.onClose(stakePoolId);
-  };
-
-  handleOpen = (event: SyntheticMouseEvent<HTMLElement>) => {
-    const {
-      onClose,
-      onClick,
-      onHover,
-      isHighlighted,
-      stakePool,
-      containerClassName,
-    } = this.props;
-    if (isHighlighted) return onClose();
-    event.persist();
-    const targetElement = event.target;
-    if (targetElement instanceof HTMLElement) {
-      const { top, left } = getRelativePosition(
-        targetElement,
-        `.${containerClassName}`
-      );
-      this.setState({ top, left });
-      if (onHover) {
-        this.handleHover(stakePool.id);
-      } else if (onClick) {
-        onClick(stakePool.id);
-      }
-    }
-    return false;
-  };
-
-  handleSelect = () => {
-    const { stakePool, onSelect } = this.props;
-    onSelect(stakePool.id);
-  };
-
-  render() {
-    const {
-      currentTheme,
-      isHighlighted,
-      isSelected,
-      onClose,
-      onHover,
-      onOpenExternalLink,
-      showWithSelectButton,
-      showSelected,
-      stakePool,
-      containerClassName,
-      numberOfRankedStakePools,
-      disabledStakePoolId,
-    } = this.props;
-    const { top, left } = this.state;
-
-    const { ranking, id } = stakePool;
-    const color = getColorFromRange(ranking, numberOfRankedStakePools);
-    const isDisabled = disabledStakePoolId === id;
+}) {
+  const { isSelected, numberOfRankedStakePools, stakePool } = props;
+  const { ranking, id } = stakePool;
+  const color = getColorFromRange(ranking, numberOfRankedStakePools);
+  const isDisabled = props.disabledStakePoolId === id;
+  const [isHighlighted, setIsHighlighted] = useState(false);
 
-    const contentClassnames = classnames([
-      styles.content,
-      isDisabled ? styles.disabled : null,
-      isSelected && showSelected ? styles.isSelected : null,
-      isHighlighted ? styles.isHighlighted : null,
-      onHover ? styles.isOnHover : null,
-    ]);
+  const contentClassnames = classnames([
+    styles.content,
+    isDisabled ? styles.disabled : null,
+    isHighlighted ? styles.isHighlighted : null,
+    props.highlightOnHover ? styles.isOnHover : null,
```

Comment:
Maybe `isHovered` or just `hovered` would be better than `isOnHover`

Reply:
I will rename this to `shouldHighlightOnHover` – this is not "the hover state" but rather if this style is added it sets up a `:hover` selector which in turn handles the "is hovered" automatically via css

Reply:
👍